src/main/java/John.java

```diff
@@ -0,0 +1,141 @@
+import java.util.Scanner;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+public class John {
+    private static final List<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        command(new String[] {"start"});
+        try (Scanner listen = new Scanner(System.in)) {
+            while (true) {
+                String line = listen.nextLine().trim();
+                if (line.isEmpty()) continue;
+                String[] cmd = line.split("\\s+");
+                command(cmd);
+                if (cmd[0].equals("bye")) break;
+            }
+        }
+    }
+
+    private static Integer parseIndex(String token) {
+        try {
+            int idx = Integer.parseInt(token) - 1; // user-facing 1-based
+            if (idx < 0 || idx >= tasks.size()) return null;
+            return idx;
+        } catch (NumberFormatException e) {
+            return null;
+        }
+    }
+
+    public static void command(String[] cmd) {
+        try {
+            switch (cmd[0]) {
```

Comment:
Instead of catching ArrayIndexOutOfBoundsException, you may validate the array length at the start to prevent the exception from being thrown and makes the code clearer.
```suggestion
            if (cmd.length == 0) {
                // Handle the case where cmd is empty gracefully
            }
            switch (cmd[0]) {
```